unit_testing/coding-challange-yaml.test.js

```diff
@@ -0,0 +1,202 @@
+/***************************************************
+This file was created by Billy Timimi.
+Other contributors include: *add you your name here if you were kind enough to help.*
+
+The purpose of this file is to do unit testing of all the files in the
+_CodingChallenges directory, in order to insure they are all valid.
+
+If you are trying to read the source of the website and are unsure as to what
+this file does/what unit testing is, I reccomend you check out the coding train's
+video on it: https://www.youtube.com/watch?v=S3QwafQEvSs, it's where I learn't
+everything needed to create this script.
+***************************************************/
+// import libraries.
+const yaml = require('js-yaml'); // js-yaml helps us read yaml of files
+const fs = require('fs'); // fs is file system package
+
+// if verbose set to true script will log almost everything it does.
+let verbose = false;
+
+// define all functions used later
+
+/*
+TODO: clear up this discription
+
+checkHasUndefinedReccursive() checks if an object has undefined or null
+properties, and if it doesn't but it has properties which are objects,
+recursivly check the properties to see if the have an undefined property.
+
+Is used to check if YAML has undeined feilds.
+*/
+function checkHasUndefinedReccursive(value) {
+  if (value == null) { //this checks for undefined as undefined == null is true
+    return true;
+  }
+  // If value is an object, test all of it's properties.
+  if (value instanceof Object) {
+    accumilator = false;
+    for (let key in value) {
+      accumilator = accumilator || checkHasUndefinedReccursive(value[key]);
+    }
+    return accumilator;
+  }
+  // if value is not an object or null or undefined (i.e. an int), return false.
+  return false;
+}
+
+// Create variable with path to _CodingChallenges directory
+let CodingChallangesDir = "../_CodingChallenges";
+
+// test that  script can read _CodingChallanges directory with fs.readdirSync
+// as it will imediatly throw an error if can't read directory (fs.readdir is
+// asyncronous and so test will pass despite not being able to read directory).
+test('Check that script can read _CodingChallanges directory.', () => {
+  let files = fs.readdirSync(CodingChallangesDir);
+  if (verbose) {
+    console.log("_CodingChallanges directory successfully read, here are all the files:");
+    files.forEach((file)=>console.log("\t" + file));
```

Comment:
I prefer spaces around `=>` I guess I should add some sort of linter to this repo? Is AirBnB style guide what I should adopt and is there a linter that goes with it? This is new to me, but I would love to make a video about it so advice appreciated! @meiamsome

Reply:
I also need to work on style, would love to see a video about it.